changes.go

```diff
@@ -20,15 +20,30 @@ type Logger interface {
 	Tracef(string, ...interface{})
 }
 
+// ArchConstraint defines an architecture constraint. This is used to represent
+// a parsed application architecture constraint.
+type ArchConstraint interface {
```

Comment:
Small semantics nit: if it implements the interface, it should have an arch constraint so the `HasArch` would be redundant. Also, returning a `*string` to emulate if-it-has-one semantics is a bit odd. How about using something like:

```go
type ConstraintGetter interface {
  // Returns the arch from the constraint or an error satisfying errors.IsNotFound
  // if the constraint does not include an arch component.  
  Arch() (string, error)
}
```

Reply:
Yeah, I could do that, I just wanted to pass in the constraints.Parse function.

I'll change, it is a bit confusing.